### ISSUE TITLE
SlevomatCodingStandard.Variables.UselessVariable.UselessVariable: false positive

### ISSUE THREAD

**Issue by jrfnl** [slevomat/coding-standard#1427]
### Given the following code sample:
```php
function foo($backtrace) {
    for ($class = $backtrace[$frame]['class']; ($parent = get_parent_class($class)) !== false; $class = $parent);
    return $class;
}
```

### With the following command:
```bash
phpcs -ps ./test.php --standard=SlevomatCodingStandard --sniffs=SlevomatCodingStandard.Variables.UselessVariable
```

###  Generates the following output:
```
-------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------
 4 | ERROR | [x] Useless variable $class.
   |       |     (SlevomatCodingStandard.Variables.UselessVariable.UselessVariable)
-------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------------------
```

### And the code would be fixed to:
```php
function foo($backtrace) {
    for ($class = $backtrace[$frame]['class']; ($parent = get_parent_class($class)) !== false; return $parent);
}
```

### While I expected

* CS: No error.
* CBF: The actual logic of the code not to be changed.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/a52720f4de29237f64fca8f9bacf00705497460f

**Comment by jrfnl** [comment]
Thanks for fixing this @kukulich !
